dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -1,25 +1,101 @@
-// Copyright 2023 ChainSafe Systems (ON)
+// Copyright 2024 ChainSafe Systems (ON)
 // SPDX-License-Identifier: LGPL-3.0-only
 
-package parachain
+package candidatevalidation
 
 import (
 	"bytes"
+	"context"
 	"errors"
 	"fmt"
+	"sync"
 
 	parachainruntime "github.com/ChainSafe/gossamer/dot/parachain/runtime"
 	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/internal/log"
 	"github.com/ChainSafe/gossamer/lib/common"
-
 	"github.com/ChainSafe/gossamer/pkg/scale"
 )
 
+var logger = log.NewFromGlobal(log.AddContext("pkg", "parachain-candidate-validation"))
+
 var (
 	ErrValidationCodeMismatch   = errors.New("validation code hash does not match")
 	ErrValidationInputOverLimit = errors.New("validation input is over the limit")
 )
 
+type CandidateValidation struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+
+	SubsystemToOverseer chan<- any
+	OverseerToSubsystem <-chan any
+}
+
+func (cv *CandidateValidation) Run(ctx context.Context, OverseerToSubSystem chan any, SubSystemToOverseer chan any) {
+	cv.wg.Add(1)
+	go cv.processMessages()
+}
+
+func (*CandidateValidation) Name() parachaintypes.SubSystemName {
+	return parachaintypes.CandidateValidation
+}
+
+func (*CandidateValidation) ProcessActiveLeavesUpdateSignal(signal parachaintypes.ActiveLeavesUpdateSignal) error {
+	// TODO: determine how this subsystem should process active leaves update signal and implement functionality
+	return nil
+}
+
+func (*CandidateValidation) ProcessBlockFinalizedSignal(signal parachaintypes.BlockFinalizedSignal) error {
+	// TODO: determine how this subsystem should process block finalized signal and implement functionality
```

Comment:
lets link an issue here too

Reply:
I've determined that this subsystem does not process this signal, I've updated comments.